### PR TITLE
Core Models and Validations

### PR DESCRIPTION
**Pull request by carlospazuzu**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -51,6 +51,10 @@ gem "bootsnap", require: false
 group :development, :test do
   # See https://guides.rubyonrails.org/debugging_rails_applications.html#debugging-with-the-debug-gem
   gem "debug", platforms: %i[ mri mingw x64_mingw ]
+  gem 'rspec-rails', '6.0.1'
+  gem 'shoulda', '~> 4.0'
+  gem "factory_bot", "~> 6.2"
+  gem 'factory_bot_rails'
```
Comment (luccamps) [inline Gemfile:57]: I think you don't actually need to have both factory_bot and factory_bot_rails in the gem file. You can probably remove factory_bot <3

```suggestion
  gem 'factory_bot_rails'
```

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  describe 'validations' do
+    context 'handle' do
+      it { is_expected.to validate_length_of(:handle).is_at_least(4).is_at_most(12) }
+      it { is_expected.not_to allow_value('carlo$$$').for(:handle) }
+      it { is_expected.to validate_uniqueness_of(:handle) }
+    end
```
Comment (luccamps) [inline spec/models/user_spec.rb:9]: You could also check if there's a presence validation and maybe if it allows a valid value <3

```suggestion
    context 'handle' do
      it { is_expected.to validate_presence_of(:handle) }
      it { is_expected.to allow_value('valid_user').for(:handle) }
      it { is_expected.to validate_length_of(:handle).is_at_least(4).is_at_most(12) }
      it { is_expected.not_to allow_value('carlo$$$').for(:handle) }
      it { is_expected.to validate_uniqueness_of(:handle) }
    end
```

File: app/models/medium.rb [hunk 3]
```diff
@@ -0,0 +1,3 @@
+class Medium < ApplicationRecord
+  validates :kind, presence: true
+end
```
Comment (luccamps) [inline app/models/medium.rb:3]: It's specified that medium has a kind/type/format attribute and a url attribute. You can also check if the url attribute is valid with a regex <3

```suggestion
class Medium < ApplicationRecord
  URL_REGEX = //

  validates :kind, presence: true
  validates :url, presence: true, format: { with: URL_REGEX, message: 'not a valid url' }
end
```

File: Gemfile (hunk 1 above)
Comment (carlospazuzu) [reply Gemfile:57]: Sure thing! Thanks!

File: spec/models/user_spec.rb (hunk 2 above)
Comment (carlospazuzu) [reply spec/models/user_spec.rb:9]: Yep, I forgot it! Thanks for the reminder!

File: app/models/medium.rb (hunk 3 above)
Comment (carlospazuzu) [reply app/models/medium.rb:3]: Totally forgot the url validation 😢

**Pull request merged**